src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -709,6 +709,23 @@ public static void Uri_EmptyPort_UsesDefaultPort()
             Assert.Equal(80, u.Port);
         }
 
+        [Fact]
+        public static void Uri_CombineUsesNewUriString()
+        {
+            // Tests that internal Uri fields were properly reset during a Combine operation
+            // Otherwise, the wrong Uri string would be used if the relative Uri contains non-ascii characters
+            // This will only affect parsers without the IriParsing flag - only custom parsers
+            UriParser.Register(new GenericUriParser(GenericUriParserOptions.GenericAuthority), "combine-scheme", -1);
+
+            const string RelativeUriString = "/relative/uri/with/non/ascii/\u00FC";
+
+            var absoluteUri = new Uri("combine-scheme://foo", UriKind.Absolute);
+            var relativeUri = new Uri(RelativeUriString, UriKind.Relative);
+
+            new Uri(absoluteUri, relativeUri);
+            new Uri(absoluteUri, RelativeUriString);
+        }
```

Comment:
Would it make sense to test the result Uri against the original? (e.g. ToStrings)

Reply:
I can expand the test to do that, in its current form it's testing that the ctor won't throw for these inputs as it's the simplest way to validate the change.